forte/data/converter/converter.py

```diff
@@ -0,0 +1,343 @@
+                              allow_new_hparam=True)
+
+    @staticmethod
+    def default_configs():
+        """
+        Returns a dictionary of default hyper-parameters.
+
+        .. code-block:: python
+
+            {
+                "to_numpy": True,
+                "to_torch": True
+            }
+
+        Here:
+
+            `"to_numpy"`: bool
+                Whether convert to `numpy.ndarray`.
+                Default is True.
+
+            `"to_torch"`: bool
+                 Whether convert to `torch.tensor`. Default is True.
+
+            .. note::
+                If `need_pad` in :class:`forte.data.converter.Feature`
+                is False and `to_numpy` and `to_torch` is True,
+                it will raise an exception if the target data cannot be
+                converted to `numpy.ndarray` or `torch.tensor`.
+
+        .. note::
+            If `need_pad` in :class:`forte.data.converter.Feature`
+            is True and `to_torch` is True, `to_torch` will overwrite the
+            effect of `to_numpy`.
+        """
+        return {
+            "to_numpy": True,
+            "to_torch": True
+        }
+
+    @property
+    def to_numpy(self) -> bool:
+        return self._config.to_numpy
+
+    @property
+    def to_torch(self) -> bool:
+        return self._config.to_torch
+
+    @property
+    def state(self) -> Dict:
+        return {
+            "to_numpy": self.to_numpy,
+            "to_torch": self.to_torch
+        }
+
+    def load_state(self, state: Dict):
+        self._to_numpy = state["to_numpy"]
+        self._to_torch = state["to_torch"]
+
+    def convert(self, features: List[Feature]) -> \
+            Tuple[Any, List[Any]]:
+        """
+        Convert a list of Features to actual data, where
+
+        1. The outer most dimension will always be the batch dimension (i.e
+        `len(output) = len(feature_num)`).
+
+        2. The type can be:
+
+            2.1 A `List` of primitive `int` or another `List`
+
+            2.2 A `numpy.ndarray`
+
+            2.3 A `torch.Tensor`
+
+        If `need_pad` in :class:`forte.data.converter.Feature` is True, it
+        will pad all features with given `pad_value` stored inside
+        :class:`forte.data.converter.Feature`.
+
+        If `to_numpy` is True, it will try to convert data into
+        `numpy.ndarray`.
+
+        If `to_torch` is True, it will try to convert data into
+        `torch.tensor`.
+
+        Args:
+            features (List[Feature]):
+                A list of :class:`forte.data.converter.Feature`
+
+        Returns:
+            A `Tuple` containing two elements.
+
+            1. The first element is either a `List` or `numpy.ndarray` or
+            `torch.tensor` representing the batch of data.
+
+            2. The second element is a `List` or `numpy.ndarray` representing
+            masks along different feature dimensions.
+
+        Example 1:
```

Comment:
One thing that is very helpful for such documentation is `doctest`. You don't have to add this now. But it is very helpful for future.